addOns/accessControl/src/main/java/org/zaproxy/zap/extension/accessControl/ExtensionAccessControl.java

```diff
@@ -627,4 +633,39 @@ public void importContextData(Context ctx, Configuration config) throws Configur
             }
         }
     }
+
+    protected int getScanProgress(int contextId) {
+
+        AccessControlScannerThread scannerThread = threadManager.getScannerThread(contextId);
+
+        if (scannerThread.isRunning() || scannerThread.hasRun()) {
+            double progress =
+                    scannerThread.getScanProgress()
+                            * 1.0
+                            / scannerThread.getScanMaximumProgress()
+                            * 100;
+            return (int) Math.round(progress);
+        } else {
+            log.error("Access control scan not running for context: " + contextId);
+            throw new IllegalStateException("A scan is not running for context: " + contextId);
+        }
+    }
+
+    protected String getScanStatus(int contextId) {
+
+        AccessControlScannerThread scannerThread = threadManager.getScannerThread(contextId);
```

Comment:
We should use other method, this ends up creating an `AccessControlScannerThread` unnecessarily.